src/main/java/org/traccar/storage/DatabaseStorage.java

```diff
@@ -307,14 +309,25 @@ private String formatOrder(Order order) {
             if (order.getDescending()) {
                 result.append(" DESC");
             }
-            if (order.getLimit() > 0) {
-                if (databaseType.equals("Microsoft SQL Server")) {
+        }
+        return result.toString();
+    }
+    
+    private String formatPagination(Pagination pagination) {
+        StringBuilder result = new StringBuilder();
+        if (pagination != null) {
+        	if(pagination.getSkip() > 0) {
+        		result.append(" OFFSET ");
+                result.append(pagination.getSkip());
+        	}
+            if (pagination.getLimit() > 0) {
+            	if (databaseType.equals("Microsoft SQL Server")) {
                     result.append(" OFFSET 0 ROWS FETCH FIRST ");
-                    result.append(order.getLimit());
+                    result.append(pagination.getLimit());
                     result.append(" ROWS ONLY");
                 } else {
                     result.append(" LIMIT ");
-                    result.append(order.getLimit());
+                    result.append(pagination.getLimit());
                 }
```

Comment:
Which databases have you tested?

Reply:
PG so far

Reply:
We need to test at least H2, MySQL, Postgres and MS SQL. Is there an SQL standard for it?

Reply:
code wise, the path is correct?

Reply:
I only did a brief look, but I think it makes sense. I would separate database changes and report changes though.